File: src/test/java/com/github/nsc/de/compiler/interpreter/values/NullValueTest.java
package com.github.nsc.de.compiler.interpreter.values;

import org.junit.jupiter.api.Test;

import static com.github.nsc.de.compiler.interpreter.values.NullValue.*;
import static com.github.nsc.de.compiler.interpreter.values.BooleanValue.*;
import static org.junit.jupiter.api.Assertions.*;

public class NullValueTest {

    @Test
    public void testAdd() {
        Error error = assertThrows(Error.class, () -> NULL.add(NULL));
        assertEquals("Operator '+' is not defined for type null", error.getMessage());
    }

    @Test
    public void testSub() {
        Error error = assertThrows(Error.class, () -> NULL.sub(NULL));
        assertEquals("Operator '-' is not defined for type null", error.getMessage());
    }

    @Test
    public void testMul() {
        Error error = assertThrows(Error.class, () -> NULL.mul(NULL));
        assertEquals("Operator '*' is not defined for type null", error.getMessage());
    }

    @Test
    public void testDiv() {
        Error error = assertThrows(Error.class, () -> NULL.div(NULL));
        assertEquals("Operator '/' is not defined for type null", error.getMessage());
    }

    @Test
    public void testPow() {
        Error error = assertThrows(Error.class, () -> NULL.pow(NULL));
        assertEquals("Operator '**' is not defined for type null", error.getMessage());
    }



    @Test
    public void testOr() {
        Error error = assertThrows(Error.class, () -> NULL.or(NULL));
        assertEquals("Operator '||' is not defined for type null", error.getMessage());
    }

    @Test
    public void testAnd() {
        Error error = assertThrows(Error.class, () -> NULL.and(NULL));
        assertEquals("Operator '&&' is not defined for type null", error.getMessage());
    }



    @Test
    public void testEqualsEquals() {
        assertEquals(TRUE, NULL.equals(NULL));
        Error error = assertThrows(Error.class, () -> NULL.equals(FALSE));
        assertEquals("Operator '==' is not defined for null and boolean", error.getMessage());
    }

    @Test
    public void testBiggerEquals() {
        Error error = assertThrows(Error.class, () -> NULL.bigger_equals(NULL));
        assertEquals("Operator '>=' is not defined for type null", error.getMessage());
    }

    @Test
    public void testSmallerEquals() {
        Error error = assertThrows(Error.class, () -> NULL.smaller_equals(NULL));
        assertEquals("Operator '<=' is not defined for type null", error.getMessage());
    }

    @Test
    public void testBigger() {
        Error error = assertThrows(Error.class, () -> NULL.bigger(NULL));
        assertEquals("Operator '>' is not defined for type null", error.getMessage());
    }

    @Test
    public void testSmaller() {
        Error error = assertThrows(Error.class, () -> NULL.smaller(NULL));
        assertEquals("Operator '<' is not defined for type null", error.getMessage());
    }



}

File: src/test/kotlin/com/github/nsc/de/shake/interpreter/values/NullValueTest.kt
package com.github.nsc.de.shake.interpreter.values

import org.junit.jupiter.api.Test
import org.junit.jupiter.api.Assertions
import java.lang.Error

class NullValueTest {
    @Test
    fun testAdd() {
        val error = Assertions.assertThrows(Error::class.java) { NullValue.NULL.add(NullValue.NULL) }
        Assertions.assertEquals("Operator '+' is not defined for type null", error.message)
    }

    @Test
    fun testSub() {
        val error = Assertions.assertThrows(Error::class.java) { NullValue.NULL.sub(NullValue.NULL) }
        Assertions.assertEquals("Operator '-' is not defined for type null", error.message)
    }

    @Test
    fun testMul() {
        val error = Assertions.assertThrows(Error::class.java) { NullValue.NULL.mul(NullValue.NULL) }
        Assertions.assertEquals("Operator '*' is not defined for type null", error.message)
    }

    @Test
    fun testDiv() {
        val error = Assertions.assertThrows(Error::class.java) { NullValue.NULL.div(NullValue.NULL) }
        Assertions.assertEquals("Operator '/' is not defined for type null", error.message)
    }

    @Test
    fun testMod() {
        val error = Assertions.assertThrows(Error::class.java) { NullValue.NULL.mod(NullValue.NULL) }
        Assertions.assertEquals("Operator '%' is not defined for type null", error.message)
    }

    @Test
    fun testPow() {
        val error = Assertions.assertThrows(Error::class.java) { NullValue.NULL.pow(NullValue.NULL) }
        Assertions.assertEquals("Operator '**' is not defined for type null", error.message)
    }

    @Test
    fun testOr() {
        val error = Assertions.assertThrows(Error::class.java) { NullValue.NULL.or(NullValue.NULL) }
        Assertions.assertEquals("Operator '||' is not defined for type null", error.message)
    }

    @Test
    fun testAnd() {
        val error = Assertions.assertThrows(Error::class.java) { NullValue.NULL.and(NullValue.NULL) }
        Assertions.assertEquals("Operator '&&' is not defined for type null", error.message)
    }

    @Test
    fun testEquals() {
        Assertions.assertEquals(BooleanValue.TRUE, NullValue.NULL.equals(NullValue.NULL))
        Assertions.assertEquals(BooleanValue.FALSE, NullValue.NULL.equals(BooleanValue.FALSE))
    }

    @Test
    fun testBiggerEquals() {
        val error = Assertions.assertThrows(Error::class.java) { NullValue.NULL.bigger_equals(NullValue.NULL) }
        Assertions.assertEquals("Operator '>=' is not defined for type null", error.message)
    }

    @Test
    fun testSmallerEquals() {
        val error = Assertions.assertThrows(Error::class.java) { NullValue.NULL.smaller_equals(NullValue.NULL) }
        Assertions.assertEquals("Operator '<=' is not defined for type null", error.message)
    }

    @Test
    fun testBigger() {
        val error = Assertions.assertThrows(Error::class.java) { NullValue.NULL.bigger(NullValue.NULL) }
        Assertions.assertEquals("Operator '>' is not defined for type null", error.message)
    }

    @Test
    fun testSmaller() {
        val error = Assertions.assertThrows(Error::class.java) { NullValue.NULL.smaller(NullValue.NULL) }
        Assertions.assertEquals("Operator '<' is not defined for type null", error.message)
    }
}